keyboards/star75/keymaps/via/keymap.c

```diff
@@ -0,0 +1,97 @@
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+	LAYOUT_all(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS)
+};
+
+/* capslock/layer indicators; using rgb led; start */
+// capslock;
+const rgblight_segment_t PROGMEM my_capslock_layer[] = RGBLIGHT_LAYER_SEGMENTS(
+    {0, 1, HSV_RED}
+);
+// layer 1;
+const rgblight_segment_t PROGMEM my_layer1_layer[] = RGBLIGHT_LAYER_SEGMENTS(
+    {0, 1, HSV_CYAN}
+);
+// layer 2;
+const rgblight_segment_t PROGMEM my_layer2_layer[] = RGBLIGHT_LAYER_SEGMENTS(
+    {0, 1, HSV_PURPLE}
+);
+// layer 3;
+const rgblight_segment_t PROGMEM my_layer3_layer[] = RGBLIGHT_LAYER_SEGMENTS(
+    {0, 1, HSV_GREEN}
+);
+
+// Now define the array of layers. Later layers take precedence
+const rgblight_segment_t* const PROGMEM my_rgb_layers[] = RGBLIGHT_LAYERS_LIST(
+    my_capslock_layer,
+    my_layer1_layer,    // Overrides caps lock layer
+    my_layer2_layer,    // Overrides other layers
+    my_layer3_layer     // Overrides other layers
+);
+
+void keyboard_post_init_user(void) {
+    // Enable the LED layers
+    rgblight_layers = my_rgb_layers;
+}
+
+bool led_update_user(led_t led_state) {
+    rgblight_set_layer_state(0, led_state.caps_lock);
+    return true;
+}
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    rgblight_set_layer_state(1, layer_state_cmp(state, 1));
+    rgblight_set_layer_state(2, layer_state_cmp(state, 2));
+    rgblight_set_layer_state(3, layer_state_cmp(state, 3));
+    return state;
+}
+/* capslock/layer indicators; using rgb led; end */
+
+#ifdef ENCODER_ENABLE
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) {
+        if (clockwise) {
+                tap_code16(dynamic_keymap_get_keycode(biton32(layer_state), 5, 3));
+        } else {
+                tap_code16(dynamic_keymap_get_keycode(biton32(layer_state), 5, 4));
+        }
+    }
+    return false;
+}
+#endif
```

Comment:
This doesn't work well.  It only supports basic keycodes, and modded basic keycodes. 

encoder map works better, but via doesn't actually support them (even tough it supports the dynamic keymap functionality. 

If you must have the encoders, then the "encoder action" stuff may be a better option.

Reply:
yes sir, they are meant to support basic keycodes. may i know where i can look up for encoder action stuff?

Reply:
https://github.com/qmk/qmk_firmware/blob/aa06893b58ac3ed88e796968d237f60f4c156875/keyboards/work_louder/work_board/keymaps/via/keymap.c
https://github.com/qmk/qmk_firmware/blob/aa06893b58ac3ed88e796968d237f60f4c156875/keyboards/work_louder/encoder_actions.c

Ideally, the default layout should not include the additional spots for the encoder, and the via keymap would.

Reply:
thanks a lot for the helps. i will try and see if i can get the via keymap to comply. the current code is working, because i am trying to leverage on VIA for the users, including key assignments for the encoder.

for the default keymap, can i put KC_TRNS for K53 and K54 instead?

Reply:
welcome!

And honestly, ideally, the default keymap (and layout macro) shouldn't have a spot for the encoders.  You can specify a via specific layout macro in the keymap.c file, and handle it that way.  (which is what the above linked keymap does, actually)

Reply:
thanks for the clarifications. i think i got the wrong ideas about what to change. 

can i keep the codes in bool encoder_update_user() in the current via keymap.c? i'm not sure if i can write the right ones with encoder_actions